### ISSUE TITLE
lint/correctness/noUnusedVariables falsePositive for multiple `new` overloads with generic parameters

### ISSUE THREAD

**Issue by GiuseppeIII** [biomejs/biome#8435]
### Environment information

```bash
...
```

### Rule name

lint/correctness/noUnusedVariables

### Playground link

https://biomejs.dev/playground/?attributePosition=multiline&code=dAB5AHAAZQAgAFQAZQBzAHQAIAA9ACAAewAKAAkAbgBlAHcAIAA8AFQAPgAoACkAOgAgAFQAOwAKAAkAbgBlAHcAIAA8AFQAPgAoAHsAIAB0AGUAcwB0ADoAIABUACAAfQApADoAIABUADsACgB9AA%3D%3D

### Expected result

```
type Test = {
	new <T>(): T;
	new <T>({ test: T }): T;
}
```

Currently gives `This type parameter T is unused.` one the first `new` line. 
As the generic parameter is used there, I would expect it to work correctly. 
(Note: identical overloads without the `new` keyword work fine
```
type Test = {
  	<T>(): T;
	<T>({ test: T }): T;
}
```
gives no issue
)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ruidosujeira** [comment]
The issue is that `TS_CONSTRUCT_SIGNATURE_TYPE_MEMBER` is missing from scope handling in `crates/biome_js_semantic/src/events.rs`, while `TS_CALL_SIGNATURE_TYPE_MEMBER` is present. 

Adding it to both enter and leave methods should fix this. 

I can open a PR.

**Comment by dyc3** [comment]
@ruidosujeira Go for it
